diff -u nginx-1.6.0/src/http/modules/ngx_http_autoindex_module.c nginx-1.6.0/src/http/modules.new/ngx_http_autoindex_module.c
--- nginx-1.6.0/src/http/modules/ngx_http_autoindex_module.c	2014-04-24 21:52:25.000000000 +0900
+++ nginx-1.6.0/src/http/modules.new/ngx_http_autoindex_module.c	2014-05-08 22:35:06.528627453 +0900
@@ -40,6 +40,7 @@
     ngx_flag_t     enable;
     ngx_flag_t     localtime;
     ngx_flag_t     exact_size;
+    ngx_flag_t     show_dot_files;
 } ngx_http_autoindex_loc_conf_t;
 
 
@@ -81,6 +82,13 @@
       offsetof(ngx_http_autoindex_loc_conf_t, exact_size),
       NULL },
 
+    { ngx_string("autoindex_show_dot_files"),
+      NGX_HTTP_MAIN_CONF|NGX_HTTP_SRV_CONF|NGX_HTTP_LOC_CONF|NGX_CONF_FLAG,
+      ngx_conf_set_flag_slot,
+      NGX_HTTP_LOC_CONF_OFFSET,
+      offsetof(ngx_http_autoindex_loc_conf_t, show_dot_files),
+      NULL },
+
       ngx_null_command
 };
 
@@ -279,9 +287,11 @@
 
         len = ngx_de_namelen(&dir);
 
-        if (ngx_de_name(&dir)[0] == '.') {
-            continue;
-        }
+	if (!alcf->show_dot_files) {
+        	if (ngx_de_name(&dir)[0] == '.') {
+            		continue;
+        	}
+	}
 
         if (!dir.valid_info) {
 
@@ -413,8 +423,13 @@
 
     b->last = ngx_cpymem(b->last, "</h1>", sizeof("</h1>") - 1);
 
-    b->last = ngx_cpymem(b->last, "<hr><pre><a href=\"../\">../</a>" CRLF,
-                         sizeof("<hr><pre><a href=\"../\">../</a>" CRLF) - 1);
+    if (!alcf->show_dot_files) {
+        b->last = ngx_cpymem(b->last, "<hr><pre><a href=\"../\">../</a>" CRLF,
+                             sizeof("<hr><pre><a href=\"../\">../</a>" CRLF) - 1);
+    } else {
+        b->last = ngx_cpymem(b->last, "<hr><pre>" CRLF,
+                             sizeof("<hr><pre>" CRLF) - 1);
+    }
 
     tp = ngx_timeofday();
 
@@ -667,6 +682,7 @@
     conf->enable = NGX_CONF_UNSET;
     conf->localtime = NGX_CONF_UNSET;
     conf->exact_size = NGX_CONF_UNSET;
+    conf->show_dot_files = NGX_CONF_UNSET;
 
     return conf;
 }
@@ -681,6 +697,7 @@
     ngx_conf_merge_value(conf->enable, prev->enable, 0);
     ngx_conf_merge_value(conf->localtime, prev->localtime, 0);
     ngx_conf_merge_value(conf->exact_size, prev->exact_size, 1);
+    ngx_conf_merge_value(conf->show_dot_files, prev->show_dot_files, 0);
 
     return NGX_CONF_OK;
 }
Common subdirectories: nginx-1.6.0/src/http/modules/perl and nginx-1.6.0/src/http/modules.new/perl
